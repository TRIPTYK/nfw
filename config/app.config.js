"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Express = require("express");
var BodyParser = require("body-parser");
var Morgan = require("morgan");
var Cors = require("cors");
var Compression = require("compression");
var Passport = require("passport");
var ErrorHandler = require("errorhandler");
var ExpressValidator = require("express-validator");
var ServiceErrorHandler = require("./../api/services/error-handler");
var Strategies = require("./passport.config");
var Router = require('./../api/routes/v1');
var _a = require('./environment.config'), HTTPLogs = _a.HTTPLogs, api = _a.api, env = _a.env, environments = _a.environments;
/**
 * Instanciate Express application
 */
var app = Express();
/**
 * GZIP compression
 */
app.use(Compression());
/**
 * Public resources
 */
app.use(Express.static('public'));
/**
 * Expose body on req.body
 */
app.use(BodyParser.json());
app.use(BodyParser.urlencoded({ extended: true }));
/**
 * Enable CORS - Cross Origin Resource Sharing
 */
app.use(Cors());
/**
 * Passport configuration
 */
app.use(Passport.initialize());
Passport.use('jwt', Strategies.jwt);
Passport.use('facebook', Strategies.facebook);
Passport.use('google', Strategies.google);
/**
 * Set validation engine
 */
app.use(ExpressValidator());
/**
 * Set Router(s) on paths
 */
app.use("/api/" + api, Router);
/**
 * Request logging with Morgan
 * dev : console | production : file
 */
app.use(Morgan(HTTPLogs));
/**
 * Errors handlers
 */
if (env.toUpperCase() === environments.DEVELOPMENT) {
    app.use(ErrorHandler({ log: ServiceErrorHandler.notify }));
}
else {
    app.use(ErrorHandler({ log: ServiceErrorHandler.log }));
}
/**
 * Exports Express
 */
module.exports = app;
//# sourceMappingURL=app.config.js.map