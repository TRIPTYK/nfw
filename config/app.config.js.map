{"version":3,"file":"app.config.js","sourceRoot":"","sources":["app.config.ts"],"names":[],"mappings":";;AAAA,iCAAmC;AAEnC,wCAA0C;AAC1C,+BAAiC;AACjC,2BAA6B;AAC7B,yCAA2C;AAC3C,mCAAqC;AACrC,2CAA6C;AAC7C,oDAAsD;AACtD,qEAAuE;AACvE,8CAAgD;AAEhD,IAAM,MAAM,GAAG,OAAO,CAAC,oBAAoB,CAAC,CAAC;AAEvC,IAAA,oCAAsE,EAApE,sBAAQ,EAAE,YAAG,EAAE,YAAG,EAAE,8BAAgD,CAAC;AAE7E;;GAEG;AACH,IAAM,GAAG,GAAG,OAAO,EAAE,CAAC;AAEtB;;GAEG;AACH,GAAG,CAAC,GAAG,CAAE,WAAW,EAAE,CAAE,CAAC;AAEzB;;GAEG;AACH,GAAG,CAAC,GAAG,CAAE,OAAO,CAAC,MAAM,CAAC,QAAQ,CAAC,CAAE,CAAC;AAEpC;;GAEG;AACH,GAAG,CAAC,GAAG,CAAE,UAAU,CAAC,IAAI,EAAE,CAAE,CAAC;AAC7B,GAAG,CAAC,GAAG,CAAE,UAAU,CAAC,UAAU,CAAE,EAAE,QAAQ,EAAG,IAAI,EAAE,CAAE,CAAE,CAAC;AAExD;;GAEG;AACH,GAAG,CAAC,GAAG,CAAE,IAAI,EAAE,CAAE,CAAC;AAElB;;GAEG;AACH,GAAG,CAAC,GAAG,CAAE,QAAQ,CAAC,UAAU,EAAE,CAAE,CAAC;AAEjC,QAAQ,CAAC,GAAG,CAAC,KAAK,EAAE,UAAU,CAAC,GAAG,CAAC,CAAC;AACpC,QAAQ,CAAC,GAAG,CAAC,UAAU,EAAE,UAAU,CAAC,QAAQ,CAAC,CAAC;AAC9C,QAAQ,CAAC,GAAG,CAAC,QAAQ,EAAE,UAAU,CAAC,MAAM,CAAC,CAAC;AAE1C;;GAEG;AACH,GAAG,CAAC,GAAG,CAAE,gBAAgB,EAAE,CAAE,CAAC;AAE9B;;GAEG;AACH,GAAG,CAAC,GAAG,CAAC,UAAQ,GAAK,EAAE,MAAM,CAAC,CAAC;AAE/B;;;GAGG;AACH,GAAG,CAAC,GAAG,CAAE,MAAM,CAAC,QAAQ,CAAC,CAAE,CAAC;AAE5B;;GAEG;AACH,IAAG,GAAG,CAAC,WAAW,EAAE,KAAK,YAAY,CAAC,WAAW,EACjD;IACE,GAAG,CAAC,GAAG,CAAE,YAAY,CAAE,EAAE,GAAG,EAAG,mBAAmB,CAAC,MAAM,EAAE,CAAE,CAAC,CAAC;CAChE;KAED;IACE,GAAG,CAAC,GAAG,CAAE,YAAY,CAAE,EAAE,GAAG,EAAG,mBAAmB,CAAC,GAAG,EAAE,CAAE,CAAC,CAAC;CAC7D;AAED;;GAEG;AACH,MAAM,CAAC,OAAO,GAAG,GAAG,CAAC"}