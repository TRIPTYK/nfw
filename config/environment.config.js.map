{"version":3,"file":"environment.config.js","sourceRoot":"","sources":["environment.config.ts"],"names":[],"mappings":";;AAAA,+BAAiC;AAEjC;;;GAGG;AACH,IAAM,YAAY,GAAG,EAAE,WAAW,EAAG,aAAa,EAAG,OAAO,EAAG,SAAS,EAAE,UAAU,EAAG,YAAY,EAAE,CAAC;AAEtG,IAAM,WAAW,GAAG,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,IAAI,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,KAAK,OAAO,IAAI,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,IAAI,YAAY,CAAC,QAAQ,CAAC,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,CAAC,WAAW,EAAE,CAAC,KAAK,CAAC,CAAC,CAAC,CAAC,CAAC,OAAO,CAAC,IAAI,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,aAAa,CAAC;AAEvL,MAAM,CAAC,MAAM,CAAE,EAAE,IAAI,EAAM,OAAO,CAAC,GAAG,EAAE,SAAI,WAAW,SAAM,EAAE,CAAE,CAAC;AAElE,MAAM,CAAC,OAAO,GAAG;IACf,GAAG,EAAE,OAAO,CAAC,GAAG,CAAC,QAAQ;IACzB,YAAY,EAAE,YAAY;IAC1B,IAAI,EAAE,OAAO,CAAC,GAAG,CAAC,IAAI;IACtB,SAAS,EAAE,OAAO,CAAC,GAAG,CAAC,UAAU;IACjC,qBAAqB,EAAE,OAAO,CAAC,GAAG,CAAC,sBAAsB;IACzD,KAAK,EAAE;QACL,GAAG,EAAE,OAAO,CAAC,GAAG,CAAC,QAAQ,KAAK,MAAM;YAClC,CAAC,CAAC,OAAO,CAAC,GAAG,CAAC,eAAe;YAC7B,CAAC,CAAC,OAAO,CAAC,GAAG,CAAC,SAAS;KAC1B;IACD,GAAG,EAAE,OAAO,CAAC,GAAG,CAAC,WAAW;IAC5B,IAAI,EAAE,OAAO,CAAC,GAAG,CAAC,QAAQ,KAAK,YAAY,CAAC,CAAC,CAAC,UAAU,CAAC,CAAC,CAAC,aAAa;IACxE,QAAQ,EAAE,OAAO,CAAC,GAAG,CAAC,QAAQ,KAAK,YAAY,CAAC,CAAC,CAAC,YAAY,CAAC,CAAC,CAAC,KAAK;CACvE,CAAC"}