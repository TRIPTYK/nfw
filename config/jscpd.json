{
    "pattern": [
        "{src,tests}/**/*.ts"
    ],
    "ignore": [
        "**/node_modules/**",
        "**/coverage/**",
        "**/migrations/**",
        "**/dist/**"
    ],
    "ignorePattern": [
        "import .* from '.*';"
    ],
    "ignoreCase": true,
    "minLines": 1,
    "mode": "mild",
    "gitignore": true,
    "absolute": true
}