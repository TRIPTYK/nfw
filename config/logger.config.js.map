{"version":3,"file":"logger.config.js","sourceRoot":"","sources":["logger.config.ts"],"names":[],"mappings":";AAAA;;;;GAIG;;AAEH,iCAAmC;AAEnC,IAAM,MAAM,GAAG,OAAO,CAAC,YAAY,CAAC;IAClC,KAAK,EAAE,MAAM;IACb,MAAM,EAAE,OAAO,CAAC,MAAM,CAAC,IAAI,EAAE;IAC7B,UAAU,EAAE;QACV,EAAE;QACF,oEAAoE;QACpE,qDAAqD;QACrD,EAAE;QACF,IAAI,OAAO,CAAC,UAAU,CAAC,IAAI,CAAC,EAAE,QAAQ,EAAE,gBAAgB,EAAE,KAAK,EAAE,OAAO,EAAE,CAAC;QAC3E,IAAI,OAAO,CAAC,UAAU,CAAC,IAAI,CAAC,EAAE,QAAQ,EAAE,mBAAmB,EAAE,CAAC;KAC/D;CACF,CAAC,CAAC;AAEH,EAAE;AACF,wEAAwE;AACxE,gEAAgE;AAChE,EAAE;AACF,IAAI,OAAO,CAAC,GAAG,CAAC,QAAQ,KAAK,YAAY,EAAE;IACzC,MAAM,CAAC,GAAG,CAAC,IAAI,OAAO,CAAC,UAAU,CAAC,OAAO,CAAC,EAAE,MAAM,EAAE,OAAO,CAAC,MAAM,CAAC,MAAM,EAAE,EAAE,CAAC,CAAC,CAAC;CACjF;AAED,MAAM,CAAC,MAAM,GAAG;IACd,KAAK,EAAE,UAAC,OAAO;QACb,MAAM,CAAC,IAAI,CAAC,OAAO,CAAC,IAAI,EAAE,CAAC,CAAC;IAC9B,CAAC;CACF,CAAC;AAEF,MAAM,CAAC,OAAO,GAAG,MAAM,CAAC"}