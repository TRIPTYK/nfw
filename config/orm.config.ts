import { container } from '@triptyk/nfw-core';
import type { ConfigurationService, Env } from '../src/api/services/configuration.service.js';
import { ConfigurationServiceImpl } from '../src/api/services/configuration.service.js';

const configService = container.resolve<ConfigurationService<Env>>(ConfigurationServiceImpl);
configService.load();

export default {
  host: configService.get('DATABASE_HOST'),
  entitiesTs: ['src/database/models'],
  entities: ['dist/src/database/models'],
  user: configService.get('DATABASE_USER'),
  password: configService.get('DATABASE_PASSWORD'),
  dbName: configService.get('DATABASE_NAME'),
  port: configService.get('DATABASE_PORT'),
  type: 'mysql',
  debug: configService.get('DEBUG')
};
