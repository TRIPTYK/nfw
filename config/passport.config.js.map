{"version":3,"file":"passport.config.js","sourceRoot":"","sources":["passport.config.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,iBAkDqD;;AAlDrD,6CAAuD;AACvD,qDAAuD;AACvD,gEAAkE;AAElE,IAAI,GAAG,OAAO,CAAC,4BAA4B,CAAC,CAAC;AAG7C,6CAA0C;AAE1C,IAAM,UAAU,GAAG;IACjB,WAAW,EAAE,SAAS;IACtB,cAAc,EAAE,yBAAU,CAAC,wBAAwB,CAAC,QAAQ,CAAC;CAC9D,CAAC;AAGF;;;;;GAKG;AACH,IAAM,GAAG,GAAG,UAAO,OAAO,EAAE,IAAI;;;;;;gBAEf,qBAAM,IAAI,CAAC,QAAQ,CAAC,OAAO,CAAC,GAAG,CAAC,EAAA;;gBAAvC,IAAI,GAAG,SAAgC;gBAC7C,IAAI,IAAI;oBAAE,sBAAO,IAAI,CAAC,IAAI,EAAE,IAAI,CAAC,EAAC;gBAClC,sBAAO,IAAI,CAAC,IAAI,EAAE,KAAK,CAAC,EAAC;;;gBAGzB,sBAAO,IAAI,CAAC,OAAK,EAAE,KAAK,CAAC,EAAC;;;;KAE7B,CAAC;AAEF;;;;GAIG;AACH,IAAM,KAAK,GAAG,UAAA,OAAO,IAAI,OAAA,UAAO,KAAK,EAAE,IAAI;;;;;;gBAEtB,qBAAM,aAAa,CAAC,OAAO,CAAC,CAAC,KAAK,CAAC,EAAA;;gBAA9C,QAAQ,GAAG,SAAmC;gBACvC,qBAAM,IAAI,CAAC,UAAU,CAAC,QAAQ,CAAC,EAAA;;gBAAtC,IAAI,GAAG,SAA+B;gBAC5C,sBAAO,IAAI,CAAC,IAAI,EAAE,IAAI,CAAC,EAAC;;;gBAGxB,sBAAO,IAAI,CAAC,KAAG,CAAC,EAAC;;;;KAEpB,EATwB,CASxB,CAAC;AAEF,OAAO,CAAC,GAAG,GAAG,IAAI,uBAAW,CAAC,UAAU,EAAE,GAAG,CAAC,CAAC;AAC/C,OAAO,CAAC,QAAQ,GAAG,IAAI,cAAc,CAAC,KAAK,CAAC,UAAU,CAAC,CAAC,CAAC;AACzD,OAAO,CAAC,MAAM,GAAG,IAAI,cAAc,CAAC,KAAK,CAAC,QAAQ,CAAC,CAAC,CAAC"}