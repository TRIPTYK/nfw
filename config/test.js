const port = 8001;

export default {
  jwt: {
    secret: 'h8566MNQ18oo5cMmHROVh8566MNQ18oo5cMmHROVh8566MNQ18oo5cMmHROVh8566MNQ18oo5cMmHROV',
    accessExpires: 5,
    refreshExpires: 1440,
    iss: `http://localhost:${port}`,
    audience: `http://localhost:${port}`,

  },
  logger: {
    logToConsole: false,
    dir: 'dist/logs/',
  },
  baseURL: '/api/v1',
  port,
  database: {
    host: 'localhost',
    user: 'root',
    password: 'test123*',
    database: 'nfw_test',
    type: 'mysql',
    port: 3306,
    debug: false,
  },
  env: 'test',
  cors: {
    origin: 'http://localhost:4200',
  },
};
