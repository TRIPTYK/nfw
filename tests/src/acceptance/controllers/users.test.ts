import { DatabaseSeeder } from 'app/database/seeders/development/development.seeder.js';
import { setupIntegrationTest } from 'tests/utils/setup-integration-test.js';
import { beforeAll, expect, test } from 'vitest';
import { accessTokenAdmin } from '../../../utils/access-token.js';
import { fetchApi } from '../../../utils/config.js';

beforeAll(async () => {
  await setupIntegrationTest(DatabaseSeeder);
})

test('GET / returns list of users', async () => {
  const users = await fetchApi('users', {
    headers: {
      Authorization: accessTokenAdmin
    }
  });

  const json = await users.json();

  expect(users.status).toStrictEqual(200);
  expect(json).toMatchSnapshot();
});
