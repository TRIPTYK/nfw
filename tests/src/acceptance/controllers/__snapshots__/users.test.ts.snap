// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GET / returns list of users 1`] = `
{
  "data": [
    {
      "attributes": {
        "firstName": "amaury",
      },
      "id": "12345678910abcdef",
      "links": {
        "self": "/api/v1/users/12345678910abcdef",
      },
      "type": "users",
    },
    {
      "attributes": {
        "firstName": "sebastien",
      },
      "id": "9876543210",
      "links": {
        "self": "/api/v1/users/9876543210",
      },
      "type": "users",
    },
  ],
  "jsonapi": {
    "version": "1.0",
  },
  "links": {
    "self": "/api/v1/users",
  },
}
`;
